File: spring-boot-dubbo/spring-boot-dubbo-dao/src/main/java/com/hhjian/dao/UserRepository.java
package com.hhjian.dao;

import com.hhjian.domain.User;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

/**
 * <p>User Repository</p>
 *
 * @author <a href="mailto:hhjian.top@qq.com">hhjian</a>
 * @since 2017.09.18
 */
@Repository
public interface UserRepository extends MongoRepository<User, String> {
}

File: spring-boot-kotlin/src/main/kotlin/com/hhjian/kotlin/repository/UserRepository.kt
package com.hhjian.kotlin.repository

import com.hhjian.kotlin.domain.User
import org.springframework.data.mongodb.repository.MongoRepository

/**
 * <p>User Repository</p>
 *
 * @author <a href="mailto:hhjian.top@qq.com">hhjian</a>
 * @since 2018.01.05
 */
interface UserRepository : MongoRepository<User, String> {
    fun findByName(name: String): User
}
